test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/util/BitSetJSTest.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util
+
+import java.nio.{ByteBuffer, LongBuffer}
+import java.util.BitSet
+import org.junit.Assert.{assertThrows => junitAssertThrows, _}
+import org.junit.Test
+import org.scalajs.testsuite.utils.AssertThrows._
+import org.scalajs.testsuite.niobuffer.SupportsTypedArrays
+import org.scalajs.testsuite.utils.Platform._
+
+object BitSetJSTest extends SupportsTypedArrays
+
+class BitSetJSTest {
+
+  // This test fails in BitSetTest.scala without the SupportsTypedArrays
+  @Test def valueOf_ByteBuffer(): Unit = {
+    val eightBS = makeEightBS()
```

Comment:
@sjrd should I keep this separate file, or is there some other remediation? or just delete this file all together?

It fails with:

```
[error] Test org.scalajs.testsuite.javalib.util.BitSetTest.valueOf_ByteBuffer failed: scala.scalajs.js.JavaScriptException: ReferenceError: Int8Array is not defined, took 0.002881443 sec
[error]     at java.nio.TypedArrayByteBuffer$.allocate(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:8781)
[error]     at org.scalajs.testsuite.javalib.util.BitSetTest.valueOf_ByteBuffer(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:127908)
[error]     at org.scalajs.testsuite.javalib.util.BitSetTest$scalajs$junit$bootstrapper$.invokeTest(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:213593)
[error]     at <jscode>.{anonymous}()(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:193804)
[error]     at org.scalajs.junit.JUnitTask.executeTestMethod(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:193852)
[error]     at org.scalajs.junit.JUnitTask.runTests$1(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:194078)
[error]     at <jscode>.{anonymous}()(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:194082)
[error]     at scala.scalajs.runtime.AnonFunction1.apply(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:311018)
[error]     at <jscode>.{anonymous}()(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:283578)
[error]     at scala.scalajs.runtime.AnonFunction1.apply(/Users/ericpeters/even/scala-js/test-suite/js/.2.12/target/scala-2.12/scalajs-test-suite-test-fastopt/main.js:311018)
[error] Failed: Total 35, Failed 1, Errors 0, Passed 34
[error] Failed tests:
[error] 	org.scalajs.testsuite.javalib.util.BitSetTest
```

Reply:
Since it is only one test, and it only fails at run-time, it is best to include it normally with the rest of `BitSetTest.scala`, with an `assumeTrue` at the beginning to ignore it when typed arrays are not supported.

Normally that would be
```scala
// at the top:
import org.scalajs.testsuite.utils.Platform._

...

  @Test def valueOf_ByteBuffer(): Unit = {
    assumeTrue("requires support for direct Buffers", typedArrays)

    ...
  }
```
But `typedArrays` itself is not available when compiling on the JVM, obviously.

I think the correct thing to do is to *add* a
```scala
def hasDirectBuffers: Boolean = typedArrays
```
in `test-suite/js/.../Platform.scala`, and
```scala
def hasDirectBuffers: Boolean = true
```
in `test-suite/jvm/.../Platform.scala`.
Then to use
```scala
assumeTrue("requires support for direct Buffers", hasDirectBuffers)
```
I'm surprised we didn't have any existing test that required this so far, TBH. ;)